packages/viewer/src/scenes/operations.ts

```diff
@@ -0,0 +1,241 @@
+import { SceneItemOperationsExecutor } from './scene';
+
+interface ShowItemOperation {
+  type: 'show';
+}
+
+interface HideItemOperation {
+  type: 'hide';
+}
+
+interface SuppliedIdSelector {
+  type: 'supplied-id';
+  value: string;
+}
+
+interface ItemIdSelector {
+  type: 'item-id';
+  value: string;
+}
+
+/**
+ * A selector builder that matches at an item level.
+ */
+export class ItemSelectorBuilder implements SelectorBuilder<ItemSelector> {
+  private query?: ItemSelector;
+
+  public build(): ItemSelector {
+    if (this.query == null) {
+      throw new Error('Cannot build selector. A selector has not been defined');
+    }
+    return this.query;
+  }
+
+  public or(): SelectorBuilder<OrSelector> {
+    return new OrSelectorBuilder(this);
+  }
+
+  public and(): SelectorBuilder<AndSelector> {
+    return new AndSelectorBuilder(this);
+  }
+
+  public withSuppliedId(suppliedId: string): this {
+    this.query = { type: 'supplied-id', value: suppliedId };
+    return this;
+  }
+
+  public withItemId(itemId: string): this {
+    this.query = { type: 'item-id', value: itemId };
+    return this;
+  }
+}
+
+/**
+ * A selector builder to perform boolean `or` operations.
+ */
+export class OrSelectorBuilder implements SelectorBuilder<OrSelector> {
+  private builders?: SelectorBuilder<ItemSelector>[] = [];
+
+  private parent: SelectorBuilder<ItemSelector>;
```

Comment:
what's `parent` intended for?

Reply:
copied this from vertex-web, haven't finished this yet.